messaging/src/main/java/org/jboss/pnc/messaging/DefaultMessageSender.java

```diff
@@ -118,6 +118,8 @@ protected void doSendMessage(String message, Map<String, String> headers) {
             messageProducer = session.createProducer(destination);
             sendUsingProducer(message, headers, session, messageProducer);
         } catch (Exception e) {
+            // try to set up a new connection on exception for the next message
+            init();
```

Comment:
Shouldn't we retry after init? And throw the MessagingRuntimeException only if it fails again?

Reply:
Yeah can do. I'll try to get that change in. Thanks for the suggestion!